[Task-7] - Authorization:

AlexEPM opened the pull request:
- created new service authorization-service with corresponding structure;
   - created basicAuthorizer lambda;
   - added local env variable with user permission in required format;
   - added logic of 401 and 403 response in case of user authentication failure;
   - added authorization logic to the importFiles lambda by use basicAuthorizer;
   - added Basic Authorization to the import file logic;
   - added obtaining authorization_token from the localStorage;
   - added logic to display alerts in case of the response statuses 401 and 403 HTTP;
   
   UI link - https://d3w0txh5vyrqln.cloudfront.net
   FE PR - https://github.com/AlexEPM/shop-react-redux-cloudfront/pull/5

Commit:
[Task-7] - Authorization:
   - created new service authorization-service with corresponding structure;
   - created basicAuthorizer lambda;
   - added local env variable with user permission in required format;
   - added logic of 401 and 403 response in case of user authentication failure;
   - added authorization logic to the importFiles lambda by use basicAuthorizer;

dmytroosypiuk commented on authorization-service/src/functions/basicAuthorizer/handler.ts:
```diff
@@ -0,0 +1,70 @@
+        console.log('policy = ', JSON.stringify(policy));
+
+        callback(null, policy);
+      } catch (e) {
+        console.log(e);
+        callback(`Unauthorized ${e.message}`);
+      }
```
why try/catch? Is it just for something like "not valid token" error or something? I think validation should be in regex. Something like this. But yea it is easier than validate everything on each step.
![image](https://user-images.githubusercontent.com/94276879/227878785-3a8701a0-c6d8-4978-bd8a-ccc636b92a52.png)

AlexEPM replied:
Yes, you are right - try-catch used here for simplicity, since there are no requirements to separate validation for the token value presence as such (the value of the token after the word "Basic"), there are no rules for validating of presence of two values in the token separated by ":", etc. To simplify, for any non-valid value of the transferred token, it is considered that the authorization has not passed, which is indicated by the corresponding error.

Regarding the validation, via "idenitityValidationExpression", that's a great point, I'll use it, thank you.
However, this approach will not help to validate all necessary cases, for example, the presence of any structure in the already decoded value of the token (for example - presence of two values ​​separated by ":").

dmytroosypiuk replied:
Yes, I understand this. You need to decode this first ofc.

Commit:
[Task-7] - Authorization:
   - deleted redundant console;
   - added 'identityValidationExpression' authorization property according to the code review comments;
